### PR TITLE
Migrated to java 8. Added AQL extension.

### PR DESCRIPTION
**Pull request by alexpits**

### CODE REVIEW

File: services/src/main/groovy/org/jfrog/artifactory/client/model/xray/settings/impl/XraySettingsImpl.groovy [hunk 1]
```diff
@@ -23,17 +22,17 @@ public void setXrayIndex(Boolean xrayIndex) {
 
     @Override
     public boolean equals(Object o) {
```
Comment (eyalbe4) [inline services/src/main/groovy/org/jfrog/artifactory/client/model/xray/settings/impl/XraySettingsImpl.groovy]: A few semi-colons (;) were removed in this method.
Let's add them back.

File: services/src/main/groovy/org/jfrog/artifactory/client/model/xray/settings/impl/XraySettingsImpl.groovy [hunk 2]
```diff
@@ -23,17 +22,17 @@ public void setXrayIndex(Boolean xrayIndex) {
...
-        if (this == o) return true;
```
Comment (eyalbe4) [inline services/src/main/groovy/org/jfrog/artifactory/client/model/xray/settings/impl/XraySettingsImpl.groovy]: A few semi-colons (;) were removed by this commit.
Let's add them back.

File: services/src/main/groovy/org/jfrog/artifactory/client/model/xray/settings/impl/XraySettingsImpl.groovy [hunk 3]
```diff
@@ -23,17 +22,17 @@ public void setXrayIndex(Boolean xrayIndex) {
...
 
-        XraySettingsImpl that = (XraySettingsImpl) o;
+        XraySettingsImpl that = (XraySettingsImpl) o
 
-        return xrayIndex != null ? xrayIndex.equals(that.xrayIndex) : that.xrayIndex == null;
+        return xrayIndex != null ? xrayIndex.equals(that.xrayIndex) : that.xrayIndex == null
     }
 
     @Override
     public int hashCode() {
```
Comment (eyalbe4) [inline services/src/main/groovy/org/jfrog/artifactory/client/model/xray/settings/impl/XraySettingsImpl.groovy]: A few semi-colons (;) were removed by this commit.
Let's add them back.

File: services/src/main/java/org/jfrog/artifactory/client/aql/AqlInclude.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package org.jfrog.artifactory.client.aql;
+
+import org.apache.commons.lang3.ArrayUtils;
+
+import java.util.Arrays;
+import java.util.StringJoiner;
+
+public class AqlInclude {
```
Comment (eyalbe4) [inline services/src/main/java/org/jfrog/artifactory/client/aql/AqlInclude.java]: Some of code lines in this methods seem not be aligned (not indented).

File: services/src/main/java/org/jfrog/artifactory/client/impl/jackson/RepositorySettingsMixIn.java [hunk 5]
```diff
@@ -5,7 +5,30 @@
 import com.fasterxml.jackson.annotation.JsonTypeInfo;
```
Comment (eyalbe4) [inline services/src/main/java/org/jfrog/artifactory/client/impl/jackson/RepositorySettingsMixIn.java]: Multiple import lines have been added, since the following import line has been removed:
import org.jfrog.artifactory.client.model.repository.settings.impl.*;
I suggest we revert this change.

File: services/src/main/java/org/jfrog/artifactory/client/impl/util/Util.java [hunk 6]
```diff
@@ -5,24 +5,29 @@
 import com.fasterxml.jackson.core.JsonProcessingException;
```
Comment (eyalbe4) [inline services/src/main/java/org/jfrog/artifactory/client/impl/util/Util.java]: The changes in this class seem redundant.
I suggest reverting them.

File: services/src/test/java/org/jfrog/artifactory/client/aql/AqlQueryBuilderTest.java [hunk 7]
```diff
@@ -0,0 +1,252 @@
+package org.jfrog.artifactory.client.aql;
```
Comment (eyalbe4) [inline services/src/test/java/org/jfrog/artifactory/client/aql/AqlQueryBuilderTest.java]: This test class includes the following formatter comments multiple times:
//@formatter:off
//@formatter:on
Let's remove them.

**Comment by eyalbe4** [comment]
@alexpits,
Thank you so much for this great pull request!
I added a few inline comments.
Also, can you please resolve the  conflict in RepositorySettingsMixIn.java and add the new AQL Builder APIs to the README?
Thanks

File: services/src/main/groovy/org/jfrog/artifactory/client/model/xray/settings/impl/XraySettingsImpl.groovy (hunk 1 above)
Comment (banadiga) [reply services/src/main/groovy/org/jfrog/artifactory/client/model/xray/settings/impl/XraySettingsImpl.groovy]: added back

File: services/src/main/groovy/org/jfrog/artifactory/client/model/xray/settings/impl/XraySettingsImpl.groovy (hunk 2 above)
Comment (banadiga) [reply services/src/main/groovy/org/jfrog/artifactory/client/model/xray/settings/impl/XraySettingsImpl.groovy]: added back

File: services/src/main/java/org/jfrog/artifactory/client/aql/AqlInclude.java (hunk 4 above)
Comment (banadiga) [reply services/src/main/java/org/jfrog/artifactory/client/aql/AqlInclude.java]: fixed

File: services/src/test/java/org/jfrog/artifactory/client/aql/AqlQueryBuilderTest.java (hunk 7 above)
Comment (banadiga) [reply services/src/test/java/org/jfrog/artifactory/client/aql/AqlQueryBuilderTest.java]: removeed

**Comment by eyalbe4** [comment]
@banadiga, @@alexpits,
Can you please resolve the conflict so that I can merge your code?

**Comment by eyalbe4** [comment]
Closing this PR, since https://github.com/jfrog/artifactory-client-java/pull/214 replaces it.
